### ISSUE TITLE
OCaml 4.12 Release Readiness

### ISSUE THREAD

**Issue by dra27** [ocaml/opam-repository#17530]
This issue tracks the build and readiness information of core platform tools and libraries in preparation for the OCaml 4.12 release. It’s intended as a meta-issue / holding place for TODO lists and links to other milestones/issues rather than general discussion (which is welcomed on Discuss or on linked issues).

Packages that are fixed upstream, have a PR ready, or can be used as-is for testing purpose, are in https://github.com/kit-ty-kate/opam-alpha-repository

**Platform tools**:
* [x] opam
* [x] dune
* [x] merlin
* [x] lsp-server / vscode-plugin (https://github.com/ocaml/opam-repository/pull/18355)
* [x] odoc
* [x] ocamlformat
* [x] ocp-indent

**Ecosystem**:
* [x] ocamlfind
* [x] ocamlbuild
* [x] ppxlib
* [x] ocaml-migrate-parsetree
* [x] mdx
* [x] utop
* [x] dune-release
* [x] opam-publish

**Infrastructure**:
* [x] ocaml.org (update /docs index, see https://github.com/ocaml/ocaml.org/pull/1124)
* [x] docker base containers
* [x] ocaml-ci
* [x] datakit-ci

Issues to be fixed in the compiler first before the release:
* [x] https://github.com/ocaml/ocaml/pull/10006
* [x] https://github.com/ocaml/ocaml/issues/10011

**Comment by olafhering** [comment]
`stdlib-shims` fails `dune runtest`, with dune 2.7.0 and `dune.git#master`. Is such thing part of verification?
````
....
[   53s] Running[1]: (cd src && /usr/bin/ocaml -I +compiler-libs /home/abuild/rpmbuild/BUILD/ocaml-stdlib-shims-0.2.0/_build/.dune/default/src/dune.ml)
[   53s] Actual targets:
[   53s] - recursive alias @install
[   53s] Running[2]: (cd _build/default && /usr/bin/ocamlc.opt -w -40 -g -a -o src/stdlib_shims.cma)
[   53s] Running[3]: (cd _build/default && /usr/bin/ocamlopt.opt -w -40 -g -a -o src/stdlib_shims.cmxa)
[   53s] Running[4]: (cd _build/default && /usr/bin/ocamlopt.opt -w -40 -g -shared -linkall -I src -o src/stdlib_shims.cmxs src/stdlib_shims.cmxa)
[   53s] + exit 0
....
[   54s] Running[2]: (cd src && /usr/bin/ocaml -I +compiler-libs /home/abuild/rpmbuild/BUILD/ocaml-stdlib-shims-0.2.0/_build/.dune/default/src/dune.ml)
[   54s] File "test/dune", line 2, characters 7-11:
[   54s] 2 |  (name test)
[   54s]            ^^^^
[   54s] Error: No rule found for src/stdlib_shims.a
````

**Comment by Octachron** [comment]
I think it would be worthwhile to have add an  `ocaml-option` for the dynamic naked pointer checker in time for the release.

**Comment by XVilka** [comment]
`mdx` and `ocaml.org` items can be checked now.

**Comment by dra27** [comment]
#18021 adds `ocaml-option-nnpchecker`

**Comment by XVilka** [comment]
Should this be closed now?

**Comment by avsm** [comment]
This shouldn't have been closed yet. lsp-server is still not available for 4.12.0, and hence VSCode doesn't work with it.

**Comment by avsm** [comment]
All merged now.
